app/pages/sections/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js

```diff
@@ -0,0 +1,223 @@
+  catalogueItemId: '10000-001',
+  deliveryDate: '2020-04-27',
+  quantity: 3,
+  estimationPeriod: 'month',
+  provisioningType: 'OnDemand',
+  type: 'flat',
+  currencyCode: 'GBP',
+  itemUnit: {
+    name: 'consultation',
+    description: 'per consultation',
+  },
+  price: 0.1,
+};
+
+const selectedPriceState = ClientFunction((orderItemState) => {
+  const cookieValue = JSON.stringify(orderItemState);
+
+  document.cookie = `selectedPrice=${cookieValue}`;
+});
+
+const mocks = () => {
+  nock(orderApiUrl)
+    .get('/api/v1/orders/order-id/sections/catalogue-solutions/existing-order-id')
+    .reply(200, orderItem);
+};
+
+const pageSetup = async (withAuth = true, postRoute = false) => {
+  if (withAuth) {
+    mocks();
+    await setCookies();
+    await selectedRecipientIdState();
+    if (postRoute) {
+      await solutionNameState();
+      await serviceRecipientNameState();
+      await selectedPriceState(orderItem);
+    }
+  }
+};
+
+fixture('Catalogue-solutions - flat ondemand - general')
+  .page('http://localhost:1234/order/some-fake-page')
+  .afterEach(async (t) => {
+    const isDone = nock.isDone();
+    if (!isDone) {
+      nock.cleanAll();
+    }
+
+    await t.expect(isDone).ok('Not all nock interceptors were used!');
+  });
+
+test('should populate input fields for day, month and year if data is returned from api', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const inputFields = Selector('#deliveryDate input:not([name=_csrf])');
+  const dayInput = inputFields.nth(0);
+  const monthInput = inputFields.nth(1);
+  const yearInput = inputFields.nth(2);
+
+  await t
+    .expect(dayInput.exists).ok()
+    .expect(dayInput.getAttribute('value')).eql('27')
+    .expect(monthInput.exists).ok()
+    .expect(monthInput.getAttribute('value')).eql('04')
+    .expect(yearInput.exists).ok()
+    .expect(yearInput.getAttribute('value')).eql('2020');
+});
+
+test('should populate text field for the quantity question', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const quantity = Selector('[data-test-id="question-quantity"] input');
+
+  await t
+    .expect(quantity.exists).ok()
+    .expect(quantity.getAttribute('value')).eql('3');
+});
+
+test('should populate the selectEstimationPeriod question radio button', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const selectEstimationPeriodRadioOptions = Selector('[data-test-id="question-selectEstimationPeriod"] input').nth(0);
+
+  await t
+    .expect(selectEstimationPeriodRadioOptions.exists).ok()
+    .expect(selectEstimationPeriodRadioOptions.hasAttribute('checked')).ok();
+});
+
+test('should render the price table content', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const priceInput = Selector('[data-test-id="question-price"] input');
+
+  await t
+    .expect(priceInput.exists).ok()
+    .expect(priceInput.getAttribute('value')).eql('0.1');
+});
```

Comment:
can you also add an expectation to this test that is renders `per consultation` as the unit of order